src/city_data_cache/city_data_cache.py

```diff
@@ -20,62 +23,79 @@ def __init__(
         self.cache_directory.mkdir(parents=True, exist_ok=True)
         self.max_file_count = max_file_count
 
-    def _get_path_to_cache_for_file(self, city_id: str, date: datetime.date) -> Path:
-        return self.cache_directory / city_id / f"{date.isoformat()}.json"
-
-    def _get_path_to_city_cache(self, city_id: str) -> Path:
-        return self.cache_directory / city_id
+    def _zip_city_path(self, city_id: str) -> Path:
+        return self.cache_directory / f"{city_id}.zip"
 
     def get(self, city_id: str, date: datetime.date) -> ResponseCityData | None:
-        cache_file_path = self._get_path_to_cache_for_file(city_id, date)
-        if not cache_file_path.is_file():
+        city_zip_path = self._zip_city_path(city_id)
+        file_name = f"{date.isoformat()}.json"
+
+        if not city_zip_path.exists():
             return None
 
-        return ResponseCityData.model_validate_json(
-            cache_file_path.read_text(encoding="utf-8")
-        )
+        with ZipFile(city_zip_path, "r") as zip_file:
+            if file_name not in zip_file.namelist():
+                return None
+            with zip_file.open(file_name) as file:
+                return ResponseCityData.model_validate_json(file.read().decode("utf-8"))
```

Comment:
We can load JSON files directly from bytes (decoding to `utf-8` will be used under the hood)
```suggestion
                return ResponseCityData.model_validate_json(file.read())
```